pkg/storegateway/indexheader/index/symbols.go

```diff
@@ -41,8 +41,13 @@ type Symbols struct {
 const symbolFactor = 32
 
 // NewSymbols returns a Symbols object for symbol lookups.
-func NewSymbols(factory *streamencoding.DecbufFactory, version, offset int) (s *Symbols, err error) {
-	d := factory.NewDecbufAtChecked(offset, castagnoliTable)
+func NewSymbols(factory *streamencoding.DecbufFactory, version, offset int, doChecksum bool) (s *Symbols, err error) {
+	tab := castagnoliTable
+	if !doChecksum {
+		tab = nil
+	}
+
+	d := factory.NewDecbufAtChecked(offset, tab)
```

Comment:
[nit] Even if the logic is the same, calling `NewDecbufAtUnchecked()` when `!doChecksum` may be easier to read. An alternative with nearly the same lines of code could be:

```
var d Decbuf

if doChecksum {
  d = factory.NewDecbufAtChecked(offset, castagnoliTable)
} else {
  d = factory.NewDecbufAtUnchecked(offset)
}
```